src/canmatrix/canmatrix.py

```diff
@@ -711,15 +715,36 @@ def to_compound_integer(self):
         else:
             return self.id
 
-    def __eq__(self, other):
-        return (
-            self.id == other.id
-            and (
-                self.extended is None
-                or other.extended is None
-                or self.extended == other.extended
-            )
-        )
+    def __eq__(self, other):  # type: (ArbitrationId) -> ArbitrationId
+        return self.id == other.id and self.extended == other.extended
+
+    def __le__(self, other):
+        if not self.extended and other.extended:
+            return True
+        if self.extended and not other.extended:
+            return False
+        return self.id <= other.id
```

Comment:
```suggestion
        return (self.extended, self.id) <= (other.extended, other.id)
```

But can this not be done via some attrs configuration and we could remove `cmp=False`?

Reply:
Done